fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RuntimeFilterGenerator.java

```diff
@@ -111,8 +114,21 @@ public PhysicalPlan visitPhysicalHashJoin(PhysicalHashJoin<? extends Plan, ? ext
                         continue;
                     }
                     Slot olapScanSlot = aliasTransferMap.get(unwrappedSlot).second;
+                    long buildSideNdv = -1L;
+                    AbstractPlan right = (AbstractPlan) join.right();
+                    if (right.getStats() != null) {
+                        List<Double> ndvs = join.getHashJoinConjuncts().stream()
+                                .map(Expression::getInputSlots)
+                                .flatMap(Set::stream)
+                                .filter(s -> right.getOutputExprIdSet().contains(s.getExprId()))
+                                .map(s -> right.getStats().columnStatistics().get(s))
+                                .filter(Objects::nonNull)
+                                .map(cs -> cs.ndv)
+                                .collect(Collectors.toList());
+                        buildSideNdv = (long) StatsMathUtil.multiNdv(ndvs);
```

Comment:
buildSideNdv should be less than build side row count.
buildSideNdv = Math.max(buildSideNdv, rowCount)

Reply:
done